extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts

```diff
@@ -0,0 +1,402 @@
+		]);
+	}
+
+	public async openDialog(): Promise<void> {
+		this.initializeDialog();
+		this.dialog.okButton.label = constants.addDatabaseReferenceOkButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => await this.addReferenceClick()));
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+
+		azdata.window.openDialog(this.dialog);
+		await this.initDialogPromise;
+	}
+
+	private dispose(): void {
+		this.toDispose.forEach(disposable => disposable.dispose());
+	}
+
+	private initializeDialog(): void {
+		this.initializeTab();
+		this.dialog.content = [this.addDatabaseReferenceTab];
+	}
+
+	private initializeTab(): void {
+		this.addDatabaseReferenceTab.registerContent(async view => {
+			this.view = view;
+			const radioButtonGroup = this.createRadioButtons();
+			this.systemDatabaseFormComponent = this.createSystemDatabaseDropdown();
+			this.dacpacFormComponent = this.createDacpacTextbox();
+			const locationDropdown = this.createLocationDropdown();
+			const variableSection = this.createVariableSection();
+
+			this.formBuilder = <azdata.FormBuilder>view.modelBuilder.formContainer()
+				.withFormItems([
+					{
+						title: '',
+						components: [
+							radioButtonGroup,
+							this.systemDatabaseFormComponent,
+							locationDropdown,
+							variableSection
+						]
+					}
+				], {
+					horizontal: false
+				})
+				.withLayout({
+					width: '100%'
+				});
+
+			let formModel = this.formBuilder.component();
+			await view.initializeModel(formModel);
+
+			this.initDialogComplete?.resolve();
+		});
+	}
+
+	public async addReferenceClick(): Promise<void> {
+		let referenceSettings: ISystemDatabaseReferenceSettings | IDacpacReferenceSettings;
+
+		if (this.currentReferenceType === ReferenceType.systemDb) {
+			referenceSettings = {
+				databaseName: <string>this.databaseNameTextbox?.value,
+				systemDb: <string>this.systemDatabaseDropdown?.value === constants.master ? SystemDatabase.master : SystemDatabase.msdb
+			};
+		} else { // this.currentReferenceType === ReferenceType.dacpac
+			referenceSettings = {
+				databaseName: <string>this.databaseNameTextbox?.value,
+				databaseLocation: <DatabaseReferenceLocation>this.referenceLocationMap.get(<string>this.locationDropdown?.value),
+				dacpacFileLocation: vscode.Uri.file(<string>this.dacpacTextbox?.value),
+				databaseVariable: <string>this.databaseVariableTextbox?.value
+			};
+			// TODO: add project reference support
+		}
+
+		await this.addReference!(this.project, referenceSettings);
+
+		this.dispose();
+	}
+
+	private createRadioButtons(): azdata.FormComponent {
+		// TODO: add project reference button
+		const systemDatabaseRadioButton = this.view!.modelBuilder.radioButton()
+			.withProperties({
+				name: 'referenceType',
+				label: constants.systemDatabaseRadioButtonTitle
+			}).component();
+
+		systemDatabaseRadioButton.checked = true;
+		systemDatabaseRadioButton.onDidClick(() => {
```

Comment:
The call to the function within OnDidClick shouldn't be async?

Reply:
nope since nothing is being awaited